README.md

```diff
@@ -157,6 +188,300 @@ the next request either as a hidden form field or via JavaScript async request.
+
+```
+X-CSRF-HEADER=1
+```
+
+When handling the request, the API checks for the existence of this header. If the header does not exist, the backend rejects the request as potential forgery. Employing a custom header allows to reject [simple requests](https://developer.mozilla.org/en-US/docs/Web/HTTP/CORS#simple_requests) that browsers do not designate as "to be preflighted" and permit them to be sent to any origin.
+
+In order to enable CSRF protection you need to add `CsrfHeaderMiddleware` to your `RouteCollectorInterface` configuration
+
+```php
+return [
+    Yiisoft\Yii\Http\Application::class => [
+        '__construct()' => [
+            'dispatcher' => DynamicReference::to(static function (Injector $injector) {
+                return ($injector->make(MiddlewareDispatcher::class))
+                    ->withMiddlewares(
+                        [
+                            ErrorCatcher::class,
+                            CsrfHeaderMiddleware::class, // <-- add this
+                            Router::class,
+                        ]
+                    );
+            }),
+        ],
+    ],
+];
+```
+
+or to the routes that must be protected.
+
+```php
+return [
+    RouteCollectionInterface::class => static function (RouteCollectorInterface $collector) use ($config) {
+        $collector
+            ->middleware(CsrfHeaderMiddleware::class) // <-- add this
+            ->addGroup(Group::create(null)->routes($routes));
+
+        return new RouteCollection($collector);
+    },
+];
+```
+
+By default, `CsrfHeaderMiddleware` considers only `GET`, `HEAD`, `POST` methods as unsafe operations. Requests with other HTTP methods trigger CORS preflight and do not require CSRF header validation. You can change this behavior as follows:
+
+```php
+use Yiisoft\Csrf\CsrfHeaderMiddleware;
+use Yiisoft\Http\Method;
+
+$csrfHeaderMiddleware = $container->get(CsrfHeaderMiddleware::class);
+
+// Returns a new instance with the specified list of unsafe methods.
+$csrfHeaderMiddleware = $csrfHeaderMiddleware->withUnsafeMethods([Method::POST]);
+
+// Returns a new instance with the specified header name.
+$csrfHeaderMiddleware = $csrfHeaderMiddleware->withHeaderName('X-CSRF-PROTECTION');
+```
+
+or define the `CsrfHeaderMiddleware` configuration in the DI container:
+
+`config/web/di/csrf.php`
+
+```php
+use Yiisoft\Csrf\CsrfHeaderMiddleware;
+use Yiisoft\Http\Method;
+
+return [
+    CsrfHeaderMiddleware::class => [
+        'withUnsafeMethods()' => [[Method::POST]],
+        'withHeaderName()' => ['X-CSRF-PROTECTION'],
+    ],
+];
+```
+
+The use of a custom request header for CSRF protection is based on the CORS Protocol. Thus, you **must** configure the CORS module to allow or deny cross-origin access to the backend API.
+
+> [!IMPORTANT]
+> `CsrfHeaderMiddleware` can be used to prevent forgery of same-origin requests and requests from the list of specific origins only.
```

Comment:
```suggestion
> Warning: `CsrfHeaderMiddleware` can be used to prevent forgery of same-origin requests and requests from the list of specific origins only.
```